source/extensions/network/dns_resolver/cares/dns_impl.cc

```diff
@@ -204,6 +204,13 @@ void DnsResolverImpl::AddrInfoPendingResolution::onAresGetAddrInfoCallback(
       bool can_process_v6 =
           (!parent_.filter_unroutable_families_ || available_interfaces_.v6_available_);
 
+      int min_ttl = INT_MAX;
+      // Loop through cname and get min_ttl
+      for (const ares_addrinfo_cname* cname = addrinfo->cnames; cname != nullptr;
+           cname = cname->next) {
+        min_ttl = std::min(min_ttl, cname->ttl);
```

Comment:
Looking at the RFC 1035, Emm, what happens if the TTL is 0? 0 means we should not be cached. We may need to think about this case.

```
TTL             a 32 bit signed integer that specifies the time interval
                that the resource record may be cached before the source
                of the information should again be consulted.  Zero
                values are interpreted to mean that the RR can only be
                used for the transaction in progress, and should not be
                cached.  For example, SOA records are always distributed
                with a zero TTL to prohibit caching.  Zero values can
                also be used for extremely volatile data.
```

Reply:
Thanks for taking a look on this.

The minimum of all entries is taken so if some entry is 0, we would parse it as 0 only. I roughly remember that is handled already as part of dns cache implementation. Is it not?

The highest according to the RFC is 2^31-1 which we are starting with in `min_ttl = INT_MAX`. The ttl of atleast one record would be lesser than that. Thus I felt this just change just extends to respect ttl thoroughly in the lookup but not change what is already present.

Reply:
I see, thanks. Let's add some unit tests for 0 and max.

Reply:
ack, done.
ps. I added tests considering cares will always use int (signed int32), and dns ttls can never be negative (RFC 2181)

Reply:
Thanks for reviewing this @zuercher 

I have added change as suggested by you to check in DNSResponse to set `std::min(2^31-1, std::max(0, ttl))` so that it is always bounded conforming to the RFC.